### PR TITLE
Convert CharField to TextField

### LINKED ISSUE #532: Convert most CharField instances to TextField
There now seems to be a consensus for both pulpcore and pulp_rpm, that TextField is always better than CharField for Pulp.
I gather this is because of how PostgreSQL implements these two fields on the DB side.

For consistency pulp_deb should also follow suit.

See: https://github.com/pulp/pulp_rpm/pull/2515 as a reference.

### PR DESCRIPTION
**Pull request by hstct**
closes #532

### CODE REVIEW

File: pulp_deb/app/models/content.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class GenericContent(Content):
     TYPE = "generic"
 
     relative_path = models.TextField(null=False)
-    sha256 = models.CharField(max_length=255, null=False)
+    sha256 = models.TextField(null=False)
```
Comment (mdellweg) [inline pulp_deb/app/models/content.py:25]: I know, it's hard to keep up, but for checksum fields, we decided that the length is known exactly, so we do not expect this field to change.

File: pulp_deb/app/models/content.py [hunk 2]
```diff
@@ -38,13 +38,13 @@ class ReleaseFile(Content):
...
-    suite = models.CharField(max_length=255)
-    distribution = models.CharField(max_length=255)
+    codename = models.TextField()
+    suite = models.TextField()
+    distribution = models.TextField()
     components = models.TextField(blank=True)
     architectures = models.TextField(blank=True)
     relative_path = models.TextField()
-    sha256 = models.CharField(max_length=255)
+    sha256 = models.TextField()
```
Comment (mdellweg) [inline pulp_deb/app/models/content.py:47]: Same here

File: pulp_deb/app/models/content.py [hunk 3]
```diff
@@ -80,10 +80,10 @@ class PackageIndex(Content):
     TYPE = "package_index"
 
     release = models.ForeignKey(ReleaseFile, on_delete=models.CASCADE)
-    component = models.CharField(max_length=255)
-    architecture = models.CharField(max_length=255)
+    component = models.TextField()
+    architecture = models.TextField()
     relative_path = models.TextField()
-    sha256 = models.CharField(max_length=255)
+    sha256 = models.TextField()
```
Comment (mdellweg) [inline pulp_deb/app/models/content.py:86]: here

File: pulp_deb/app/models/content.py [hunk 4]
```diff
@@ -113,10 +113,10 @@ class InstallerFileIndex(Content):
     FILE_ALGORITHM = {"SHA256SUMS": "sha256", "MD5SUMS": "md5"}  # Are there more?
 
     release = models.ForeignKey(ReleaseFile, on_delete=models.CASCADE)
-    component = models.CharField(max_length=255)
-    architecture = models.CharField(max_length=255)
+    component = models.TextField()
+    architecture = models.TextField()
     relative_path = models.TextField()
-    sha256 = models.CharField(max_length=255)
+    sha256 = models.TextField()
```
Comment (mdellweg) [inline pulp_deb/app/models/content.py:119]: and here.

File: pulp_deb/app/models/content.py (hunk 1 above)
Comment (hstct) [reply pulp_deb/app/models/content.py:25]: Thanks for clarifying I missed that. I will revert the change for the checksum fields.

**Review by mdellweg — APPROVED** [review]
LGTM

File: pulp_deb/app/migrations/0018_auto_20220601_1254.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
+# Generated by Django 3.2.13 on 2022-06-01 12:54
```
Comment (quba42) [inline pulp_deb/app/migrations/0018_auto_20220601_1254.py:1]: Can we rename the auto-generated migration file to something more meaningful?

See the corresponding change in pulp_rpm as a reference: https://github.com/pulp/pulp_rpm/pull/2515/files

**Review by quba42 — CHANGES_REQUESTED** [review]
Everything else LGTM.

**Comment by quba42** [comment]
The failed tests appear to be related to https://github.com/pulp/pulp_deb/pull/541.

**Review by quba42 — APPROVED** [review]
Rebase to let the tests go green, then this is good to merge.

**Pull request merged**